pysat/_instrument.py

```diff
@@ -198,6 +198,8 @@ def __init__(self, platform=None, name=None, tag=None, sat_id=None,
                                  'strings, or both None.')
         else:
             # user has provided a module
+            if type(inst_module) == str:
```

Comment:
Registering by using Instrument takes a single instance of Instrument and sets a pysat wide option. While perhaps convenient, it could be unclear or unexpected. The Instrument itself also has many options. I think the register method you included should be the one way.

Reply:
Hold off on changing this. I'm rethinking the comment above. It is convenient....
Note keeping this in will require a change to the Instrument docstring.

Reply:
I have removed string registration implementation and corresponding test